src/Storage/Storage.Management/StorageAccount/RestoreAzStorageBlobRange.cs

```diff
@@ -119,47 +122,103 @@ public override void ExecuteCmdlet()
+                        throw new InvalidJobStateException(string.Format("Blob ranges restore failed with information: '{0}'.", ex.ToString()));
                     }
-
-                    AzureOperationResponse<BlobRestoreStatus> result = waitTask.Result;
-
-                    WriteObject(new PSBlobRestoreStatus(result.Body));
-
                 }
                 else
                 {
-                    BlobRestoreStatus status = this.StorageClient.StorageAccounts.BeginRestoreBlobRanges(
-                     this.ResourceGroupName,
-                     this.StorageAccountName,
-                     this.TimeToRestore,
-                     PSBlobRestoreRange.ParseBlobRestoreRanges(this.BlobRestoreRange));
-
-                    WriteObject(new PSBlobRestoreStatus(status));
-                    if (status != null && status.Status == BlobRestoreProgressStatus.Failed)
+                    Track2.StorageAccountResource account = this.StorageClientTrack2.GetStorageAccount(this.ResourceGroupName, this.StorageAccountName);
+                    var restoreLro = account.RestoreBlobRanges(
+                        WaitUntil.Started,
+                        new Track2Models.BlobRestoreContent(
+                            this.TimeToRestore.ToUniversalTime(),
+                            PSBlobRestoreRange.ParseBlobRestoreRanges(this.BlobRestoreRange)));
+                    Dictionary<string, object> temp = restoreLro.GetRawResponse().Content.ToObjectFromJson() as Dictionary<string, object>;
+                    object jobStatus = null;
+
+                    if (temp != null && temp.TryGetValue("status", out jobStatus))
                     {
-                        throw new InvalidJobStateException("Blob ranges restore failed.");
+                        if (jobStatus != null && jobStatus.ToString() == Track2Models.BlobRestoreProgressStatus.Failed.ToString())
+                        {
+                            throw new InvalidJobStateException("Blob ranges restore failed.");
+                        }
+                    } else
+                    {
+                        WriteWarning(string.Format("Could not fetch the status."));
+                    }
+
+                    if (temp == null)
+                    {
+                        throw new InvalidJobStateException("Could not fetch the Blob restore response.");
                     }
+
+                    temp.TryGetValue("restoreId", out object restoreId);
+                    temp.TryGetValue("parameters", out object parameters);
+
+                    PSBlobRestoreParameters blobRestoreParameters = new PSBlobRestoreParameters();
+                    Dictionary<string, object> paramMap = parameters as Dictionary<string, object>;
+
+                    paramMap.TryGetValue("timetoRestore", out object timeToRestore);
+                    DateTimeOffset.TryParse(timeToRestore.ToString(), out DateTimeOffset parseDate);
```

Comment:
>timeToRestore [](http://example.com/codeflow?start=44&length=13)

When `paramMap.TryGetValue` fails (e.g.: there is no "timetoRestore" field), the object is null.

Reply:
Fixed